File: week_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
void counter(char *str);




int main(void)
{
    string s = get_string("Text: ");
    counter(s);
}


void counter(char *str)
{
    int ascii_code;
    int letters = 0;
    int sentences = 0;
    int words = 1;
    float index = 0;

    for (int i = 0; i < strlen(str); i++)
    {
        ascii_code = str[i];
        if ((ascii_code >= 65 && ascii_code <= 90) || (ascii_code >= 97 && ascii_code <= 122))
        {
            letters++;
        }
        if ((ascii_code == 46) || (ascii_code == 33) || (ascii_code == 63))
        {
            sentences++;
        }
        if (ascii_code == 32)
        {
            words++;
        }
    }

    index = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences / (float) words) - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

File: week_6/readability/readability.py
# Readability

# get input about the text
s = input("Text: ")

# Vars
letters = 0
sentences = 0
words = 1
index = 0

# Counter
for i in range(len(s)):
    # Convert strings to ascii
    ascii_code = ord(s[i])

    if ((ascii_code >= 65 and ascii_code <= 90) or (ascii_code >= 97 and ascii_code <= 122)):
        letters += 1

    if ((ascii_code == 46) or (ascii_code == 33) or (ascii_code == 63)):
        sentences += 1

    if (ascii_code == 32):
        words += 1

# Calculating index of text
L = letters * 100 / words
S = sentences * 100 / words
index = round(0.0588 * L - 0.296 * S - 15.8)

# Finally outputs the results
if (index < 1):
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print("Grade", index)